Source/ASNetworkImageLoadInfo.h

```diff
@@ -0,0 +1,39 @@
+//
+//  ASNetworkImageLoadInfo.h
+//  AsyncDisplayKit
+//
+//  Created by Adlai on 1/30/18.
+//  Copyright © 2018 Facebook. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <AsyncDisplayKit/ASBaseDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSInteger, ASNetworkImageSourceType) {
+  ASNetworkImageSourceUnspecified = 0,
+  ASNetworkImageSourceSynchronousCache,
+  ASNetworkImageSourceAsynchronousCache,
+  ASNetworkImageSourceFileURL,
+  ASNetworkImageSourceDownload,
+};
+
+AS_SUBCLASSING_RESTRICTED
+@interface ASNetworkImageLoadInfo : NSObject <NSCopying>
+
+/// The type of source from which the image was loaded.
+@property (readonly) ASNetworkImageSourceType sourceType;
```

Comment:
Are these all atomic on purpose?

Reply:
Yep! I'm trying to figure out if there's a way to document a readonly property as `atomic` but get a synthesized getter that behaves like `nonatomic` (since we don't need a lock/barrier.)

For the time being, I figure simplicity and correct documentation are key. We won't spend much time in the spinlock/unfair-lock that Obj-C uses for these.